accepted/0000-selective-versions-resolutions.md

```diff
@@ -0,0 +1,303 @@
+way of doing thing, and it could be considered a bad direction to go toward.
+
+Some of the alternatives below actually take this into consideration, but are
+a bit more complex in terms of expressiveness, so were not chosen by the RFC
+submitter (see open questions below too).
+
+# Alternatives
+
+There is at least one alternative to the proposed solution, more complex but
+more expressive.
+
+## Nested dependencies resolution per dependency
+
+Starting from an example, this solution would take the following form in the
+`package.json` file:
+```json
+  "devDependencies": {
+    "@angular/cli": "1.0.3",
+    "typescript": "2.3.2"
+  },
+  "resolutions": {
+    "@angular/cli": {
+      "typescript": "2.0.2"
+    }
+  }
+```
+
+yarn would use `typescript@2.0.2` only for `@angular/cli` (so in
+`node_modules/@angular/cli/node_modules`), but keep `typescript@2.3.2` in
+`node_modules/typescript`.
+
+Basically, this enables the user to specify versions for nested dependencies,
+but only in the context of a given dependency.
+
+The fields of the `resolutions` field must only refer to existing entries in
+`devDependencies` and `dependencies`.
+
+Of course, if the same version of a nested dependency is used for many
+dependencies, yarn will behave as always by keeping it directly in
+`node_modules`.
+
+## Mapping version specifications
+
+This is a kind of simplified solution to the "out-of-scope scenario" in the
+motivations section above (it maps versions but not dependency names).
+
+It was proposed in this
+[comment](https://github.com/yarnpkg/yarn/issues/2763#issuecomment-301896274).
+
+Everything is not totally clear to me, but the idea would be to map a given
+version specification to another one.
+This would take this form in the `package.json`:
+```json
+"devDependencies": {
+    "@angular/cli": "1.0.3",
+    "typescript": "2.2.2",
+    "more dependencies..."
+  },
+  "mappings": {
+    "typescript@>=2.0.0 <2.3.0": "typescript@2.3.2"
+  }
+```
+
+yarn would then replace matching version specifications with the user's one.
+What is problematic with this is that the user has to know that `@angular/cli`
+is exactly expressing its dependency to `typescript` as `>=2.0.0 <2.3.0`.
+
+This makes such mappings hard to maintain because they can become ignored if
+`@angular/cli` is upgraded and its dependency specification changes, while
+the other solutions would only result in 
+
+# Unresolved questions
+
+## Is this expressive enough?
+
+As explained in the alternative solutions section, it would be much more
+expressive and coherent with the npm ecosystem package management paradigm
+to use nested dependency resolutions per project dependency.
+Would the loss of simplicity acceptable maybe?
+
+## Warnings in logs
+
+Should yarn warn the user about an incoherence between an explicit dependency
+and a resolution. For example if the user specify a dependency to
+`typescript@2.3.2` and the resolutions field contains `typescript@2.3.0`.

```

Comment:
Considering that Yarn warns when peer dependencies don't match semver I think overridden resolutions should warn as well.
I think this is a temporary measure until the community catches up and fixes issues